sdk/eidos_sdk/io/events.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from abc import ABC
+from enum import Enum
+from pydantic import BaseModel, TypeAdapter
+from typing import List, TypeVar, Generic, Any, Dict, AsyncIterator, Type, Literal
+
+
+class StopReason(Enum):
+    ERROR = "error"
+    COMPLETED = "completed"
+    CANCELED = "canceled"
+
+
+T = TypeVar("T")
+
+
+class BaseStreamEvent(ABC, BaseModel):
+    stream_context: List[str] = None
+    event_type: Literal["start", "end", "string_output", "object_output", "tool_call", "tool_end", "error", "agent_call", "agent_end"]
+
+
+class StartStreamEvent(BaseStreamEvent):
+    event_type: str = "start"
+
+
+class EndStreamEvent(BaseStreamEvent):
+    event_type: str = "end"
+    stop_reason: StopReason
+
+
+class StringOutputEvent(BaseStreamEvent):
+    event_type: str = "string_output"
+    content: str
+
+
+class ObjectOutputEvent(BaseStreamEvent, Generic[T]):
+    event_type: str = "object_output"
+    content: T
```

Comment:
do we also want a list output? or are we happy calling this an object output event? (although lists are a little funny since you could just send events for each record or wrap it as a field of a list, which I think is fine to keep the event interface simpler for now.